leap_year.rb

```diff
@@ -0,0 +1,25 @@
+def leap_year_calc
+  puts "Give me a start year!"
+  start_year = gets.chomp
+  start_year = start_year.to_i
+  puts "Give me an ending year!"
+  end_year = gets.chomp
+  end_year = end_year.to_i
+  year = start_year
+  leap_year = Array.new
+  while year <= end_year
+  if year % 4 == 0
+    if year  % 100 == 0 && year % 400 != 0
+      break
+    elsif year % 400 == 0
+        leap_year.push(year)
+    else
+      leap_year.push(year)
+    end
+  end
+  year = year + 1
+  end
+  puts leap_year
+end
+
```

Comment:
It is working pretty well. It would be nice to have some sort of validation that ensures the user types only numbers, because, if we add any type of string like "alñkdfjldksjf", when converted into integer, it becomes 0. Then something like this:

Enter a starting year:
alñkdfjldksjf
Enter an ending year:
20

would generate

0
4
8
12
16
20

(Personal opinion) That input should return a message like "alñkdfjldksjf is not a number".